src/libostree/ostree-repo-pull.c

```diff
@@ -380,6 +344,21 @@ fetch_uri_contents_utf8_sync (OstreeFetcher  *fetcher,
 }
 
 static gboolean
+fetch_uri_contents_utf8_sync (OstreeFetcher  *fetcher,
+                              SoupURI        *uri,
+                              char          **out_contents,
+                              GCancellable   *cancellable,
+                              GError        **error)
+{
+  GSList *mirrorlist = g_slist_append (NULL, uri);
+  gboolean ret =
+    fetch_mirrorred_uri_contents_utf8_sync (fetcher, mirrorlist, NULL,
+                                            out_contents, cancellable, error);
+  g_slist_free (mirrorlist);
```

Comment:
Could use `g_autoptr(GSList) mirrorlist` and this would all condense into a single line.